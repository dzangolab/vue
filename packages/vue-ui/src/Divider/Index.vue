<template>
  <div :data-aria-orientation="orientation" class="divider" role="separator" />
</template>

<script lang="ts">
export default {
  name: "DividerElement",
};
</script>

<script setup lang="ts">
defineProps({
  orientation: {
    default: "horizontal",
    type: String,
    validator: (value: string) => ["horizontal", "vertical"].includes(value),
  },
});
</script>

<style lang="css">
.divider[role="separator"] {
  --_border: var(--divider-border, 1px solid #d0d0d0);
}

.divider[role="separator"][data-aria-orientation="horizontal"] {
  border-top: var(--_border);
  height: 0;
  margin: 1rem 0;
  width: 100%;
}

.divider[role="separator"][data-aria-orientation="vertical"] {
  border-left: var(--_border);
  height: 100%;
  margin: 0 1rem;
  width: 0;
}
</style>
