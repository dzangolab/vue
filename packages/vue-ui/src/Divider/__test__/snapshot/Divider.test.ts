import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Divider from "../../Index.vue";

describe("Divider", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Divider);

    expect(wrapper.element).toMatchSnapshot();
  });
});
