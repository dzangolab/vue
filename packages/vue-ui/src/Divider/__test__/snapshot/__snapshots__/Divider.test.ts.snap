// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > matches snapshot 1`] = `
<div
  class="divider"
  data-aria-orientation="horizontal"
  role="separator"
/>
`;
